### PR TITLE
*: add config gRPC service 

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Part of #1977. It should be reviewed after #2018 is merged.

### What is changed and how it works?
This PR adds the config gRPC service. Also it adds `EnableConfigManager` in command line to decide whether we enable configuration manager. 

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -129,6 +129,8 @@ type Config struct {
 
 	logger   *zap.Logger
 	logProps *log.ZapProperties
+
+	EnableConfigManager bool
```
Comment (disksing) [inline server/config/config.go]: I think we can enable it by default. No need the config. If no client use it, the feature has no side effect.

File: server/server.go [hunk 2]
```diff
@@ -945,5 +956,11 @@ func (s *Server) reloadConfigFromKV() error {
 		s.storage.SwitchToDefaultStorage()
 		log.Info("server disable region storage")
 	}
+
+	if s.cfg.EnableConfigManager {
+		err = s.cfgManager.Reload(s.storage)
```
Comment (overvenus) [inline server/server.go]: Does it happen before starting grpc server?

File: server/config/config.go (hunk 1 above)
Comment (rleungx) [reply server/config/config.go]: The switch will also control the PD config management which will start a goroutine in #2049.

File: server/server.go (hunk 2 above)
Comment (rleungx) [reply server/server.go]: It happened after starting the gRPC server

Comment (overvenus) [reply server/server.go]: Can we ensure it must be done before any `config.Create/Get/Update/Delete`?

File: server/config_manager/config_manager.go [hunk 3]
```diff
@@ -43,18 +45,29 @@ var (
 	errNotSupported = "not supported"
 )
 
+// Server is the interface for configuration manager.
```
Comment (shafreeck) [inline server/config_manager/config_manager.go]: Does it seem that we have another interface called `Server`? Maybe we should define the interface with some specifications.

File: server/server.go (hunk 2 above)
Comment (overvenus) [reply server/server.go]: Could you add some comments about the order of `Reload` and serving requests?

**Comment by NingLin-P** [comment]
LGTM

File: server/config_manager/config_manager.go (hunk 3 above)
Comment (rleungx) [reply server/config_manager/config_manager.go]: Yes, and actually we indeed have some interfaces called Server in different packages. Maybe we can handle it in the future. But I don't have a good idea.

**Pull request merged**